### PR TITLE
DOP-3660: Add openapi-changelog directive

### PR DESCRIPTION
**Pull request by mmeigs**
### Ticket

DOP-3660

### Notes

Adds openapi-changelog directive and api-version option.

### CODE REVIEW

File: snooty/test_parser.py [hunk 1]
```diff
@@ -3527,3 +3527,30 @@ def test_invalid_icon() -> None:
     )
     page.finish(diagnostics)
     assert len(diagnostics) == 0
+
+
+def test_openapi_changelog() -> None:
```
Comment (i80and) [inline snooty/test_parser.py:3532]: We don't actually need tests for rstspec-only changes since there are no new code paths being introduced :D

**Review by i80and — APPROVED** [review]

File: snooty/test_parser.py (hunk 1 above)
Comment (mmeigs) [reply snooty/test_parser.py:3532]: :) I was wondering! But I felt like better safe than sorry. I'll delete!

**Comment by i80and** [comment]
@mmeigs do we want this to go out to writers immediately?

**Comment by mmeigs** [comment]
> @mmeigs do we want this to go out to writers immediately or with our next release?

Oh, this is a brand new project! This is the very first ticket. This will be finished *hopefully* in about 6 weeks. So, not looking to have anyone use it. Not that they will even know about this directive until then...

**Pull request merged**